pkg/cloudprovider/providers/nifcloud/nifcloud_load_balancer.go

```diff
@@ -159,6 +177,211 @@ func (c *Cloud) EnsureLoadBalancerDeleted(ctx context.Context, clusterName strin
+		}
+
+		if ipAddress, ok := annotations[ServiceAnnotationLoadBalancerNetworkInterface1IPAddress]; ok {
+			if ipAddress != "" {
+				return fmt.Errorf("annotation %s is only enabled for %s=elb", ServiceAnnotationLoadBalancerNetworkInterface1IPAddress, ServiceAnnotationLoadBalancerType)
+			}
+		}
+
+		if ipAddress, ok := annotations[ServiceAnnotationLoadBalancerNetworkInterface2IPAddress]; ok {
+			if ipAddress != "" {
+				return fmt.Errorf("annotation %s is only enabled for %s=elb", ServiceAnnotationLoadBalancerNetworkInterface2IPAddress, ServiceAnnotationLoadBalancerType)
+			}
+		}
+
+		if systemIPAddresses, ok := annotations[ServiceAnnotationLoadBalancerNetworkInterface1SystemIPAddresses]; ok {
+			if systemIPAddresses != "" {
+				return fmt.Errorf("annotation %s is only enabled for %s=elb", ServiceAnnotationLoadBalancerNetworkInterface1SystemIPAddresses, ServiceAnnotationLoadBalancerType)
+			}
+		}
+
+		if systemIPAddresses, ok := annotations[ServiceAnnotationLoadBalancerNetworkInterface2SystemIPAddresses]; ok {
+			if systemIPAddresses != "" {
+				return fmt.Errorf("annotation %s is only enabled for %s=elb", ServiceAnnotationLoadBalancerNetworkInterface2SystemIPAddresses, ServiceAnnotationLoadBalancerType)
+			}
+		}
+
+		if vipNetwork, ok := annotations[ServiceAnnotationLoadBalancerVipNetwork]; ok {
+			if vipNetwork != "" {
+				return fmt.Errorf("annotation %s is only enabled for %s=elb", ServiceAnnotationLoadBalancerVipNetwork, ServiceAnnotationLoadBalancerType)
+			}
+		}
+	}
+	if loadBalancerType == "elb" {
+		// validation of elastic load balancer
+		if networkVolume, ok := annotations[ServiceAnnotationLoadBalancerNetworkVolume]; ok {
+			if !slices.Contains(allowedElasticLoadBalancerNetworkVolume, networkVolume) {
+				return fmt.Errorf("annotation %s=%s is invalid", ServiceAnnotationLoadBalancerNetworkVolume, networkVolume)
+			}
+		}
+
+		if policyType, ok := annotations[ServiceAnnotationLoadBalancerPolicyType]; ok {
+			if policyType != "" {
+				return fmt.Errorf("annotation %s is only enabled for %s=lb", ServiceAnnotationLoadBalancerPolicyType, ServiceAnnotationLoadBalancerType)
+			}
+		}
+
+		if proto, ok := annotations[ServiceAnnotationLoadBalancerHCProtocol]; ok {
+			if proto != "TCP" && proto != "ICMP" {
+				return fmt.Errorf("annotation %s=%s is invalid", ServiceAnnotationLoadBalancerHCProtocol, proto)
+			}
+		}
+
+		if networkInterface1, ok := annotations[ServiceAnnotationLoadBalancerNetworkInterface1]; ok {
+			if isPrivateLanNetworkID(networkInterface1) {
+				if ipAddress, ok := annotations[ServiceAnnotationLoadBalancerNetworkInterface1IPAddress]; ok {
+					if !isIPAddress(ipAddress) {
+						return fmt.Errorf("annotation %s=%s is invalid", ServiceAnnotationLoadBalancerNetworkInterface1IPAddress, ipAddress)
+					}
+				} else {
+					return fmt.Errorf("annotation %s is required when %s is private ip", ServiceAnnotationLoadBalancerNetworkInterface1IPAddress, ServiceAnnotationLoadBalancerNetworkInterface1)
```

Comment:
I think `ServiceAnnotationLoadBalancerNetworkInterface1` describes network id.

So maybe it is required when network is `private lan`.

```suggestion
					return fmt.Errorf("annotation %s is required when %s is private lan", ServiceAnnotationLoadBalancerNetworkInterface1IPAddress, ServiceAnnotationLoadBalancerNetworkInterface1)
```

Reply:
apply suggestion I applied your sggestion. 7b29f2c7b7d67f0f133f6d600e2f040174ca4402